packages/gatsby/cache-dir/query-result-store.js

```diff
@@ -91,22 +99,24 @@ export class StaticQueryStore extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      staticQueryData: getStaticQueryData(),
+      staticQueryData: { ...getStaticQueryResults() },
     }
   }
 
   handleMittEvent = () => {
     this.setState({
-      staticQueryData: getStaticQueryData(),
+      staticQueryData: { ...getStaticQueryResults() },
```

Comment:
This is just to force re-render, It doesn't need to create actual clone or even store this in state. This could probably be done with something else. On the light site - this will happen only when we load data for new page (note that this case doesn't guarantee anything with static queries changed) or what static query updates